### PR TITLE
adjust katello config to properly fetch build groups

### PR DESCRIPTION
**Pull request by evgeni**
also re-arrange inheritance to match 3.15

### CODE REVIEW

**Comment by evgeni** [comment]
(this has been executed to make https://github.com/theforeman/foreman-packaging/pull/5261 work)

File: configs/katello/3.16.yaml [hunk 1]
```diff
@@ -99,45 +98,6 @@
       - test-flat-el8
       - centos8-devel
       - puppetlabs-puppet6-rhel-8
-    build_groups:
```
Comment (evgeni) [inline configs/katello/3.16.yaml:102]: we set build_package_group_source_tag, no need to add build_groups too

Comment (ekohl) [reply configs/katello/3.16.yaml:102]: Is this something we can do in nightly and foreman as well?

Comment (evgeni) [reply configs/katello/3.16.yaml:102]: foreman yes, nightly no, as that's the source of truth.

**Pull request merged**